lib/scala/runtime-version-manager/src/main/scala/org/enso/runtimeversionmanager/components/RuntimeVersionManager.scala

```diff
@@ -704,7 +715,14 @@ class RuntimeVersionManager(
           )
         }
 
+        installRequiredRuntimeComponents(runtime, os).getOrElse {
+          throw InstallationError(
+            "fatal: Cannot install the required runtime components."
+          )
```

Comment:
This looks bad.

It throws an error but does nothing to revert the installation, so it has in fact succeeded.

I think we should either decide to warn and succeed or revert completely.

The easiest solution would be to move this *before* the `atomicMove`, so that the components are installed to the temporary runtime (because I understand that it is safe to move it, right?) and then only moved to the right place if the installation succeeds. If you move it there, throwing the error as is done right now will be enough because the temporary runtime *is* reverted on failure. The not-temporary runtime would have to be explicitly reverted, as seen above in the line `FileSystem.removeDirectory(runtimePath)`.

Reply:
Thanks! I'll just do the update in the temporary directory